mindsdb/integrations/redis/redisdb.py

```diff
@@ -0,0 +1,163 @@
+import os
+from threading import Thread
+import walrus
+from mindsdb.utilities.log import log
+from mindsdb.integrations.base import Integration
+from mindsdb.streams.redis.redis_stream import RedisStream
+from mindsdb.interfaces.storage.db import session, Stream
+
+
+class RedisConnectionChecker:
+    def __init__(self, **kwargs):
+        self.host = kwargs.get('host')
+        self.port = kwargs.get('port', 6379)
+        self.db = kwargs.get('db', 0)
+        self.user = kwargs.get('user', None)
+        self.password = kwargs.get('password', None)
+
+    def _get_connection(self):
+        return walrus.Database(host=self.host,
+                               port=self.port,
+                               db=self.db,
+                               socket_connect_timeout=10)
+
+    def check_connection(self):
+        client = self._get_connection()
+        try:
+            client.dbsize()
+            return True
+        except Exception:
+            return False
+
+
+class Redis(Integration, RedisConnectionChecker):
+    """Redis Integration which is more a Streams factory
+    than classical Integration."""
+    def __init__(self, config, name):
+        Integration.__init__(self, config, name)
+        intergration_info = self.config['integrations'][self.name]
+        self.host = intergration_info.get('host')
+        self.port = intergration_info.get('port', 6379)
+        self.db = intergration_info.get('db', 0)
+        self.control_stream_name = intergration_info.get('stream')
+        self.user = intergration_info.get('user', None)
+        self.password = intergration_info.get('password', None)
+        self.client = self._get_connection()
+        self.control_stream = self.client.Stream(self.control_stream_name)
+        self.company_id = os.environ.get('MINDSDB_COMPANY_ID', None)
+        self.streams = {}
+
+    def setup(self):
+        """Launches streams stored in db and
+        Launches a worker in separate thread which waits
+        data from control stream and creates a particular Streams."""
+
+        self.start_stored_streams()
+        # launch worker which reads control steam and spawn streams
+        self.start()
+
+    def start(self):
+        Thread(target=Redis.work, args=(self, )).start()
```

Comment:
you get access to `session` in thread, so you need close session at thread stop.

Reply:
It also be possible when we will store Integration info as separate record.

Reply:
Done